### PR TITLE
Add timeout to DeepSpeedStrategy

### LINKED ISSUE #20466: Allowing setting timeout in DeepSpeedStrategy
### Outline & Motivation

In DDPStrategy / FSDPStrategy, the `timeout=datetime.timedelta(seconds=1800)` flag is exposed and thus allowing user to tune. However, in DeepSpeedStrategy, which is a subclass of DDPStrategy, this flag is not exposed, which makes it hard to change the timeout behavior.

Is there any workaround? Otherwise, I think it might be worth adding `kwargs` to the `__init__()` function of DeepSpeedStrategy, and pass along those parameters to the parent class DDPStrategy.

### Pitch

_No response_

### Additional context

_No response_

cc @borda @awaelchli @justusschock

### PR DESCRIPTION
**Pull request by jedyang97**
## What does this PR do?

Fixes #20466. allow users to pass `kwargs` to `DeepSpeedStrategy`. This allows users to set timeout when using `DeepSpeedStrategy`.

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--20474.org.readthedocs.build/en/20474/

### CODE REVIEW

File: src/lightning/fabric/strategies/deepspeed.py [hunk 1]
```diff
@@ -97,6 +97,7 @@ def __init__(
         load_full_weights: bool = False,
         precision: Optional[Precision] = None,
         process_group_backend: Optional[str] = None,
+        **kwargs: Any,
```
Comment (lantiga) [inline src/lightning/fabric/strategies/deepspeed.py:100]: The other strategies have `timeout` as an explicit argument (https://github.com/Lightning-AI/pytorch-lightning/blob/be608fa355b835b9b0727df2f5476f0a1d90bc59/src/lightning/fabric/strategies/fsdp.py#L144).

I strongly suggest we do that here as well, adding `kwargs` hampers discoverability.

Let's keep the pattern exactly the same as FSDP.

**Review by lantiga — COMMENTED** [review]
Thanks for the PR! I added a comment

File: src/lightning/fabric/strategies/deepspeed.py (hunk 1 above)
Comment (jedyang97) [reply src/lightning/fabric/strategies/deepspeed.py]: That makes sense. I've updated the PR.

Comment (lantiga) [reply src/lightning/fabric/strategies/deepspeed.py]: looks good thanks!

**Pull request merged**